Image component

snide opened the pull request:
Work in progress on an Image component. Need to discuss with @cchaos and figure out some stuff around sizing.

cc @alexfrancoeur @nreese. This will give you what you need for the expandable screenshots in the homepage v2.

### Questions for review

1. Should `hasShadow` be on by default?
2. Should `allowFullScreen` be on by default?
3. Should a default `size` be set?

![](https://d3vv6lp55qjaqc.cloudfront.net/items/3u0k312x3Z0d1u2S3m1r/Screen%20Recording%202018-01-04%20at%2012.57%20PM.gif?X-CloudApp-Visitor-Id=59773&v=54405a9f)

cchaos commented:
Putting here mainly as a reminder:

We should be sure to require the `title` attribute on `img` tags.

nreese commented on src-docs/src/views/image/image_example.js:
```diff
@@ -0,0 +1,78 @@
+          </p>
+          <ul>
+            <li>
+              <EuiCode>size</EuiCode> accepts <EuiCode>s / m / l / original / fullWidth</EuiCode>.
```
Missing `xl` option

nreese commented on src/components/image/image.js:
```diff
@@ -0,0 +1,125 @@
+    super(props);
+
+    this.state = {
+      isImageFullscreen: false,
```
Should `isImageFullscreen` be camel-cased and be `isImageFullScreen`?

nreese commented on src/components/image/image.js:
```diff
@@ -0,0 +1,125 @@
+    this.onKeyDown = this.onKeyDown.bind(this);
+  }
+
+  onKeyDown = event => {
```
This is not passed as a prop to anything so pressing the `esc` key does not close the full screen image

snide replied:
Ha. Knew I forgot something ;-)

snide commented:
@nreese Feedback addressed and added an ESC to close mechanism.

cchaos commented on src-docs/src/views/image/image_example.js:
```diff
@@ -0,0 +1,78 @@
+          <ul>
+            <li>
+              <EuiCode>size</EuiCode> accepts <EuiCode>s / m / l / xl / original / fullWidth</EuiCode>.
+              The later will set the figure to stretch to 100% of its container.
```
sp: "later" -> "latter"

cchaos commented on src-docs/src/views/image/image_example.js:
```diff
@@ -0,0 +1,78 @@
+            </li>
+            <li>
+              <EuiCode>caption</EuiCode> will provide a caption to the image.
+            </li>
```
`title` is a prop as well

snide replied:
Addressed

cchaos commented on src-docs/src/views/image/image_example.js:
```diff
@@ -0,0 +1,78 @@
+      text: (
+        <p>
+          Images can be sized by passing the <EuiCode>size</EuiCode> prop a value
+          of <EuiCode>s / m / l / original / fullWidth</EuiCode>.
```
add 'xl' to this list as well

snide commented:
OK. Feedback addressed.

1. Gonna keep the shadow as on by default since it matches our other elements (like form elements) which have a similar small shadow. Ability is still there to turn it off.
2. `allowFullScreen` is set to off by default. Updated the doc examples.
3.  Default size set to original.

I also went ahead and added a small expand icon. This requires a new prop `fullScreenIconColor` though because the images might be dark or light and need a different color hover icon. So added that as well.

![](https://d3vv6lp55qjaqc.cloudfront.net/items/1k070K3g0e2O150r3u1e/Screen%20Recording%202018-01-05%20at%2012.56%20PM.gif?X-CloudApp-Visitor-Id=59773&v=efa1e1a9)

cchaos commented on src-docs/src/views/image/image_zoom.js:
```diff
@@ -0,0 +1,33 @@
+        allowFullScreen
+        caption="I am a starship. Zooom!"
+        title="Click me to see in full screen."
+        fullScreenIconColor="default"
```
Should this be `fullScreenIconColor="dark"` not `default`?

cchaos commented:
Oh and I just saw two more things, the second one we can either address now or when we do more in-depth documentation.

1. The `title` attribute should actually be `alt`
2. The title/alts of the image examples aren't correct alt descriptions. They should be describing the image not describing the interaction. I only mention this since we want to be sure that even our examples are fully correct so that someone who may be copy and pasting these directly from the examples will know what needs to be replaced, else they may just keep that title as is.

snide commented:
Good spots. Flipped the alts in, adjusted their text and fixed that default/dark issue. Gonna merge.

alexfrancoeur commented:
@snide woah! this is a fantastic surprise, awesome turnaround guys. @nreese is this something we can use in 6.2 or should I add to the enhancement issue I'll be creating for Kibana Home?

nreese commented:
@alexfrancoeur Its already in 6.2
